src/dapp/walletProvider.ts

```diff
@@ -285,6 +287,70 @@ export class WalletProvider implements IDappProvider {
         });
     }
 
+    /**
+     * Packs a {@link Transaction} and fetches correct redirect URL from the wallet API. Then redirects
+     *   the client to the sign transaction hook
+     * @param transaction
+     * @param options
+     */
+    async signTransaction(transaction: Transaction, options?: {callbackUrl?: string}): Promise<Transaction> {
+        if (!this.mainFrame) {
+            throw new Error("Wallet provider is not initialised, call init() first");
+        }
+
+        const {contentWindow} = this.mainFrame;
+        if (!contentWindow) {
+            throw new Error("Wallet provider is not initialised, call init() first");
+        }
+
+        return new Promise<Transaction>((resolve, reject) => {
+            let plainTransaction = transaction.toPlainObject();
```

Comment:
can you extract this into a different method, something like prepareWalletTransaction or something similar

Reply:
done